z3/src/z3_solver.cpp

```diff
@@ -0,0 +1,376 @@
+#include "z3_solver.h"
+
+#include <inttypes.h>
+
+#include <iostream>
+
+#include <z3++.h>
+// #include "z3_extensions.h"
+
+using namespace std;
+
+namespace smt {
+
+/* Z3 Op mappings */
```

Comment:
Remove these big comments for the PR. You might want to duplicate the branch to keep these and other things around.

Reply:
Related to your other comment, I think it makes sense to split out the sort features from the term features in this PR. Probably fine to leave the `NotImplementedException` stub methods though.